packages/mapper/src/mapper.ts

```diff
@@ -728,25 +919,41 @@ export default class Mapper {
    * @returns {Promise} The promise of command results. TODO: command result object
    */
   runCommand(database, cmd): Promise<any> {
-    this.messageBus.emit('method:runCommand', database._name, cmd);
-    return this.serviceProvider.runCommand(database._name, cmd);
+    const db = database._name;
+    this.messageBus.emit(
+      'mongosh:api-call',
+      { method: 'runCommand', class: 'Database', db, arguments: { cmd } }
+    );
+
+    return this.serviceProvider.runCommand(db, cmd);
   }
 
   async update(collection, filter, update, options: any = {}): Promise<any> {
+    const db = collection._database._name;
+    const coll = collection._name;
+    this.messageBus.emit(
+      'mongosh:api-call',
+      {
+        method: 'update',
+        class: 'Collection',
+        db, coll, arguments: { filter, update, options }
+      }
+    );
+
     let result;
-    this.messageBus.emit('method:update', collection._name, filter);
+
     if (options.multi) {
       result = await this.serviceProvider.updateMany(
-        collection._name,
-        collection._database._name,
+        db,
+        coll,
```

Comment:
^ so in `serviceProvider` `update` function takes `(db, collection, filter, update, options)`, what we had here before was `(collection, db, filter, update, options)`. So this little change fixes the possible bug with updates!